arches/app/models/tile.py

```diff
@@ -598,8 +614,14 @@ def get_blank_tile_from_nodegroup_id(nodegroup_id, resourceid=None, parenttile=N
         tile.resourceinstance_id = resourceid
         tile.parenttile = parenttile
         tile.data = {}
+        tile.load_serialized_graph()
```

Comment:
Because `load_serialized_graph()` does a db query, we're probably giving back all the intended performance savings. I noticed the two implementations perform roughly the same (this is with a nodegroup from AfS):


```py
python3 manage.py shell
>>> from timeit import timeit
>>> from arches.app.models.tile import Tile
>>> timeit("Tile.get_blank_tile_from_nodegroup_id('0434ccce-1d9f-11eb-a29f-024e0d439fdb')", globals=globals(), number=5000)  # pr
3.2668450839992147
>>> timeit("Tile.get_blank_tile_from_nodegroup_id('0434ccce-1d9f-11eb-a29f-024e0d439fdb')", globals=globals(), number=5000)  # dev branch
3.308565291983541
```

I can take a closer look at this tomorrow, but what do you think about that point in general, in other words only trying to do this (potential) optimization only if we already have the published graph in hand?

Reply:
The idea here is that it's likely the caller of this method will not just grab the empty tile and be finished, but that they'll set some nodes in the tile and then save it back to the database - at which point this becomes a bit more valuable because then the serialized graph doesn't have to be loaded on save.  So this is a prefetch of the the serialized graph for that tile object.  However, it is true that this is not strictly needed here, and it's probably better left for when it is actually needed - the optimization is a bit premature.